pkg/services/user/userimpl/store.go

```diff
@@ -34,6 +34,7 @@ type store interface {
 	Search(context.Context, *user.SearchUsersQuery) (*user.SearchUserQueryResult, error)
 	Count(ctx context.Context) (int64, error)
 	CountUserAccountsWithEmptyRole(ctx context.Context) (int64, error)
+	InTransaction(ctx context.Context, fn func(ctx context.Context) error) error
```

Comment:
Instead of having this I think you could just inject `db.DB` into the service to start the transaction.